pyPhin.py

```diff
@@ -109,47 +114,76 @@ def verify(self, contact, verifyUrl, deviceUUID, verificationCode):
 		'''
 		req = requests.get(
 			self.baseUrl+locationUrl,
-			headers=self.createHeader(authToken, deviceUUID, "2.0.1")
+			headers=self.createHeader(deviceUUID, authToken, "2.0.1")
 			)
 		reqJson = json.loads(req.text)
 
-		if not reqJson["success"]:
-			raise Exception(reqJson)
+		self.checkRequest(reqJson)
 
 		vesselUrl = reqJson["locations"][0]["resources"]["vessels"]["route"]
 
-		auth = {"authToken":authToken,"vesselUrl":vesselUrl,"UUID":deviceUUID}
-		return json.dumps(auth)
+		#Auth Dictionary Structure needed to access data.
+		authData = {"authToken":authToken,"vesselUrl":vesselUrl}
 
-	def createHeader(self, authToken, deviceUUID, version="1.0.0"):
-		tempHeader = {"x-phin-concise":"true",
-			"x-phin-reporting-app-id":"ios-app",
-			"x-phin-reporting-device-id":deviceUUID}
-		tempHeader["Accept-Version"] = version
-		tempHeader["Authorization"] = "Bearer " + authToken
-		return tempHeader
+		return authData
+
+	def getData(self, authToken, deviceUUID, vesselUrl):
+
+		self.checkUrlRoute(vesselUrl)
 
-	def getData(self, auth, data):
+		data = {}
 
-		dataCluster = json.loads(auth)
+		data["waterData"] = self.getWaterData(
+			authToken,
+			deviceUUID,
+			vesselUrl)
 
-		if data.upper() == "TA" or "CYA" or "TH":
-			return self.getWaterQuality(dataCluster["authToken"], dataCluster["vesselUrl"], dataCluster["UUID"], data.upper())
+		return data
 
-	def getWaterQuality(self, authToken, vesselUrl, deviceUUID, waterQualityType):
+	def getWaterData(self, authToken, deviceUUID, vesselUrl):
 
 
 		req = requests.get(
 			self.baseUrl+vesselUrl,
-			headers=self.createHeader(authToken,deviceUUID, "2.0.0")
+			headers=self.createHeader(deviceUUID, authToken, "2.0.0")
 			)
-		reqJson = json.loads(req.text)
-		print(reqJson)
-
-
-		return reqJson["vessels"][0]["waterReport"][waterQualityType]["value"]
 
+		reqJson = json.loads(req.text)
+		self.checkRequest(reqJson)
 
+		data = {}
 
+		for dataType in ["TA","CYA","TH"]:
+			data[dataType] = reqJson["vessels"][0]["waterReport"][dataType]["value"]
+		data["temperature"] = reqJson["vessels"][0]["disc"]["temperatureF"]
+		data["status"] = reqJson["vessels"][0]["disc"]["title"]
 
+		'''Sample Return data
+		{
+			"TA":100,
+			"CYA": 40,
+			"TH": 170
+			"temperature":70.9
+			"status":"balanced"
+		}
+		'''
+		return data
 
+	def createHeader(self, deviceUUID, authToken=None, version="1.0.0"):
+		headers = {"x-phin-concise":"true",
+			"x-phin-reporting-app-id":"ios-app",
+			"x-phin-reporting-device-id":deviceUUID}
+		headers["Accept-Version"] = version
+		if authToken != None:
+			headers["Authorization"] = "Bearer " + authToken
+		return headers
+
+	def checkRequest(self, json):
```

Comment:
what if "json" is not actual json.. but None or a int? first check if it's actual json